torch/optim/adam.py

```diff
@@ -278,10 +296,19 @@ def adam(params: List[Tensor],
         raise RuntimeError('torch.jit.script not supported with foreach optimizers')
 
     if fused and not torch.jit.is_scripting():
+        # lr needs to be a tensor
+        if not isinstance(lr, Tensor):
+            lr = torch.tensor(lr, dtype=torch.float)
```

Comment:
Why aren't these errors? BC of checkpoints?

Reply:
Sure, but also because people are allowed to call adam() directly.

Reply:
But this might end up on the wrong device?
In general, I would say that the minimum cross device moves code we have, the better. Unless it would be breaking current behavior, I would not enable this.

Reply:
If we don't do this, we will definitely be breaking the current behaviour for anyone calling

`adam.adam(params, grads, ..., lr: float=whatever, fused=True)`